transport-native-epoll/src/main/java/io/netty/channel/epoll/TunTapChannel.java

```diff
@@ -0,0 +1,298 @@
+/*
+ * Copyright 2014 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.netty.channel.epoll;
+
+import java.io.IOException;
+import java.net.SocketAddress;
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.List;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.ByteBufAllocator;
+import io.netty.channel.ChannelMetadata;
+import io.netty.channel.ChannelOutboundBuffer;
+import io.netty.channel.ChannelPipeline;
+import io.netty.channel.ChannelPromise;
+import io.netty.channel.unix.Socket;
+
+/**
+ * A {@link Channel} implementation that can be used to send or receive packets over
+ * a Linux TUN/TAP interface.
+ * <p>
+ * To initialize a {@link TunTapChannel}, create a {@link TunAddress} or {@link TapAddress}
+ * object containing the name of the TUN/TAP device to open, and then bind the {@link TunTapChannel}
+ * to the address.
+ * <p>
+ * To send packets into the device (causing them to be received by the local host's network
+ * stack) create a {@link TunTapPacket} object containing the packet type and data, and call
+ * {@link TunTapChannel} write() or writeAndFlush().
+ * <p>
+ * When the kernel sends packets out via the device, the packets are delivered to the channel
+ * handlers via the channelRead() method, wrapped in {@link TunTapPacket} objects.
+ */
+public class TunTapChannel extends AbstractEpollChannel {
+
+    private boolean _isOpen = true;
+    private TunTapChannelConfig _config;
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Rename this field "_config" to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS00116)